Modules/_asynciomodule.c

```diff
@@ -631,6 +631,9 @@ create_cancelled_error(FutureObj *fut)
     } else {
         exc = PyObject_CallOneArg(asyncio_CancelledError, msg);
     }
+    if (exc == NULL) {
+        return NULL;
+    }
```

Comment:
``PyException_SetContext`` steals a reference to its second argument (according to the docs), which implies that the refcount of ``fut->fut_cancelled_exc`` should be adjusted before returning (and also that the Py_CLEAR call below is dodgy). 

That said, the call to ``PyException_SetContext`` appears to be a no-op because the block at line 622 is used when that ``fut_cancelled_exc`` is not NULL and that block always returns. That means ``fut_cancelled_exc`` should be NULL if execution gets here.

Reply:
Thanks Ronald, you're 100% right. I've pushed an update.